mailet/standard/src/main/java/org/apache/james/transport/matchers/SentByMailet.java

```diff
@@ -0,0 +1,54 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+
+
+package org.apache.james.transport.matchers;
+
+import java.util.Collection;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.GenericMatcher;
+
+/**
+ * <P>Matches mails that are sent by an SMTP authenticated user.</P>
+ * <P>If the sender was not authenticated it will not match.</P>
+ * <PRE><CODE>
+ * &lt;mailet match=&quot;SMTPAuthSuccessful&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * </CODE></PRE>
+ *
+ * @version CVS $Revision$ $Date$
+ * @since 2.2.0
+ */
+public class SentByMailet extends GenericMatcher {
+    
+    /**
+     * The mail attribute holding the SMTP AUTH user name, if any.
+     */
+
+    public Collection<MailAddress> match(Mail mail) {
+        String authUser = (String) mail.getAttribute(Mail.SENT_BY_MAILET);
+        if (authUser != null) {
+            return mail.getRecipients();
+        } else {
```

Comment:
For the else yes it is not mandatory but I think it make the intention more clear